### PR TITLE
feat(api): fault injection made more generic

### PR DESCRIPTION
**Pull request by dsavitskiy**
Fault injection support moved from nexus API to a new testing API, and made more generic.
Injectiosn removed from v0 API.

### CODE REVIEW

File: protobuf/v1/testing.proto [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/empty.proto";
+import "common.proto";
+
+service TestingRpc {
+  // Fault injection.
+  rpc AddInjection (AddInjectionRequest) returns (google.protobuf.Empty) {}
+  rpc RemoveInjection (RemoveInjectionRequest) returns (google.protobuf.Empty) {}
+  rpc ListInjections (ListInjectionsRequest) returns (ListInjectionsReply) {}
```
Comment (tiagolobocastro) [inline protobuf/v1/testing.proto]: Do you suggest we add WipeReplica here?

File: protobuf/v1/testing.proto [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+message Injection {
+  string uri = 1;
+  string device_name = 2;
```
Comment (tiagolobocastro) [inline protobuf/v1/testing.proto]: Do we need both here?

File: protobuf/v1/testing.proto [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  bool is_active = 3;
```
Comment (tiagolobocastro) [inline protobuf/v1/testing.proto]: What does active mean?

File: protobuf/v1/testing.proto (hunk 1 above)
Comment (dsavitskiy) [reply protobuf/v1/testing.proto]: Yes

File: protobuf/v1/testing.proto (hunk 2 above)
Comment (dsavitskiy) [reply protobuf/v1/testing.proto]: URI defines all the parameters of an injection: target device, offset, time, fault type, etc.
Device name is just device part of it. It is given here for convenience, e.g. for CLI.
We'll probably want to add all the param here, once injections will be more or less stable.

File: protobuf/v1/testing.proto (hunk 3 above)
Comment (dsavitskiy) [reply protobuf/v1/testing.proto]: It means a condition that triggers this particular fault injection has been met, and error are being injected.

**Review by tiagolobocastro — APPROVED** [review]

File: protobuf/v1/testing.proto (hunk 2 above)
Comment (tiagolobocastro) [reply protobuf/v1/testing.proto]: I see, thanks for clarifying.

**Review by chriswldenyer — APPROVED** [review]
Looks reasonable.

**Pull request merged**